pkg/registrar/registrar.go

```diff
@@ -137,6 +137,20 @@ func registerSupplyChainController(mgr manager.Manager) error {
 		return fmt.Errorf("watch: %w", err)
 	}
 
+	mapper := Mapper{
+		Client: mgr.GetClient(),
+		Logger: mgr.GetLogger().WithName("supply-chain"),
+	}
+
+	for _, template := range v1alpha1.ValidSupplyChainTemplates {
+		if err := ctrl.Watch(
+			&source.Kind{Type: template},
+			handler.EnqueueRequestsFromMapFunc(mapper.TemplateToSupplyChainRequests),
+		); err != nil {
+			return fmt.Errorf("watch template (%v): %w", template, err)
```

Comment:
I don't think `%v` is a good choice here - it'll produce a "stringified" output that's not really that helpful

e.g.,

```go
          template := &v1alpha1.ClusterSourceTemplate{}
          panic(fmt.Errorf("watch template (%v): %w", template, errors.New("foo")))
```

leads to

```
panic: watch template (&{{ } {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {{nil  []}  } {}}): foo

goroutine 1 [running]:
main.main()
        ...main.go:13 +0xa8
```

Reply:
no what we want is the type name, but reflection returns errors of it's own... what a rabbit hole.

Reply:
As a side note, the root cause of us missing the low-quality here is the complete absence of useful testing for registrar.